### PR TITLE
Add ui_microphone

### PR DESCRIPTION
**Pull request by knolleary**
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Proposed changes

Adds `node-red-node-ui-microphone` - a single button widget to capture audio in the dashboard and send as a WAV encoded buffer.

### CODE REVIEW

**Comment by dceejay** [comment]
Cool. Are you using Built on browser mediaRecorder ? Do we need polyfills for safari and edge ?

**Comment by knolleary** [comment]
Yes, uses MediaRecorder.

Edge already supports it and it can be enabled in Safari by the user.

Here's a video of it in use with IBM Watson Speech to Text - you can simply post the output of the microphone node straight to the STT service and get the text back: https://www.youtube.com/watch?v=CxiwO4nxSgI

Lots of other bells and whistles that could be added - but this is MVP to get something published for an immediate need.

Possible future features:

 - timeslice the recording so we send X second-long chunks whilst still recording. Got this 90% working already.
 - add optional silence detection to automatically stop recording - found a couple good blog posts on how to do this.
 - add more optional UI elements - such as a time count
 - I'm sure others will think of things that they can PR

**Comment by knolleary** [comment]
Needless to say, this is my first UI node. The UI part is very simple, but angular is not something I've used much before. Hopefully its simple enough to be good enough :)

File: node-red-node-ui-microphone/ui_microphone.html [hunk 1]
```diff
@@ -0,0 +1,127 @@
+<!--
+  Copyright 2018, Seth350
```
Comment (dceejay) [inline node-red-node-ui-microphone/ui_microphone.html]: Is this correct ?

Comment (knolleary) [reply node-red-node-ui-microphone/ui_microphone.html]: oops, thought I'd tidied them up. Fix pushed.

**Pull request merged**